autoflip_calculators/text_detection_calculator_test.cc

```diff
@@ -0,0 +1,229 @@
+  const std::vector<Packet>& output_packets =
+      runner->Outputs().Tag(kOutputRegion).packets;
+  ASSERT_EQ(1, output_packets.size());
+
+  const auto& regions = output_packets[0].Get<DetectionSet>();
+  SalientRegion text;
+  ASSERT_EQ(text_labels.size() <= regions.detections().size(), true);
+  
+  // // It is better to use visiual check for the bouding boxs.
+  // for (int i = 0; i < regions.detections().size(); ++i) {
+  //     text = regions.detections(i);
+  //     EXPECT_EQ(text.signal_type().standard(), SignalType::TEXT);
+  //     EXPECT_FLOAT_EQ(abs(text.location_normalized().x()-top_left_corners[i].x) < 0.05, true);
+  //     EXPECT_FLOAT_EQ(abs(text.location_normalized().y()-top_left_corners[i].y) < 0.05, true);
+  // }
+}
+
+// TODO(zzhencchen) Add a unit test to show frames.
+// void ShowFrames(const std::vector<cv::Point2f>& top_left_corners,
+//               const std::vector<std::string>& text_labels, const float font_scale,
+//               const cv::Scalar font_color, CalculatorRunner* runner) {
+//   ASSERT_EQ(top_left_corners.size(), text_labels.size());
+//   auto input_frame =
+//       ::absl::make_unique<ImageFrame>(ImageFormat::SRGB, kImagewidth, kImageheight);
+//   cv::Mat frame = mediapipe::formats::MatView(input_frame.get());
+//   for (int i = 0; i < top_left_corners.size(); ++i) {
+//     cv::Point2f top_left_corner = top_left_corners[i];
+//     top_left_corner.x *= kImagewidth;
+//     top_left_corner.y *= kImageheight;
+//     cv::putText(frame, text_labels[i], top_left_corner, cv::FONT_HERSHEY_SIMPLEX, font_scale, font_color);
+//   }
+
+//   const std::vector<Packet>& output_packets =
+//       runner->Outputs().Tag(kOutputRegion).packets;
+//   const auto& regions = output_packets[0].Get<DetectionSet>();
+//   SalientRegion text;
+//   for (int i = 0; i < regions.detections().size(); ++i) {
+//     text = regions.detections(i);
+//     cv::Rect2f box(text.location_normalized().x(),text.location_normalized().y(), 
+//               text.location_normalized().width(), text.location_normalized().height());
+//     cv::Point2f vertices[4];
+//     box.points(vertices);
+
+//     for (int j = 0; j < 4; ++j) {
+//         vertices[j].x *= kImagewidth;
+//         vertices[j].y *= kImageheight;
+//     }
+
+//     for (int j = 0; j < 4; ++j) {
+//         line(frame, vertices[j], vertices[(j+1) % 4], Scalar(0, 255, 0), 1);
+
+//     std::string label = format("%.2f", confidency);
+//     putText(frame, label, vertices[0], FONT_HERSHEY_SIMPLEX, 0.8, Scalar(0, 0, 255));
+//     }
+//   }
+//   cv::namedWindow(kWinName);
+//   cv::imshow(kWinName, frame);
+
+//   input_frame.release();
+// }
+
+// Checks that calculator works when there is no model path.
+TEST(TextDetectionCalculatorTest, NoModelPath) {
+  auto runner = ::absl::make_unique<CalculatorRunner>(MakeConfig(kConfig, ""));
+  SetInputs(kTopLeftCornersOne, kTextLabelsOne, kFontScale, kFontColor, runner.get());
+  MP_ASSERT_OK(runner->Run());
```

Comment:
I'm confused here, wouldn't the result be NOT okay if the model path is missing:
MP_ASSERT_FALSE(runner->Run().ok())

Reply:
I see. That is what I want.